src/lib/titled-hoc.jsx

```diff
@@ -36,10 +36,14 @@ const TitledHOC = function (WrappedComponent) {
                 const defaultProjectTitle = this.handleReceivedProjectTitle();
                 this.props.onUpdateProjectTitle(defaultProjectTitle);
             }
-            // if the projectTitle hasn't changed, but the reduxProjectTitle
-            // HAS changed, we need to report that change to the projectTitle's owner
-            if (this.props.reduxProjectTitle !== prevProps.reduxProjectTitle &&
-                this.props.reduxProjectTitle !== this.props.projectTitle) {
+            // if the user has changed the project Title, and the projectTitle prop
+            // passed in to us is stale (defined, but old), we need to notify
+            // the source of the projectTitle that it should be updated
+            const reduxTitleHasChanged =
+                (this.props.reduxProjectTitle !== prevProps.reduxProjectTitle &&
+                this.props.reduxProjectTitle !== this.props.projectTitle);
+            const titlePropIsSet = !!this.props.projectTitle;
```

Comment:
Currently we allow you to set your title to an empty string — would this break that? I think it would be safer to test for `null` or `undefined`.  As an aside, I thought our eslint rules disallowed `!!`.

Reply:
Good point. Changed to test for null or undefined.